salt/core/network/init.sls

```diff
@@ -0,0 +1,61 @@
+# Define order so that these configuration options are run before everything else.
+# Other packages and commands rely on the hostname field being configured.
+
+# Don't run on old/existing servers prior to network configuration.
+{%- if 'network' in pillar %}
+{{ pillar.network.ipv4.primary_ip }}:
+  host.only:
+    - order: 5
+    - hostnames:
+      - {{ pillar.host_id }}.open-contracting.org
+      - {{ pillar.host_id }}
+{%- if 'ipv6' in pillar.network %}
+{{ pillar.network.ipv6.primary_ip }}:
+  host.only:
+    - order: 5
+    - hostnames:
+      - {{ pillar.host_id }}.open-contracting.org
+      - {{ pillar.host_id }}
+{% endif %}
+
+/etc/mailname:
+  file.managed:
+    - order: 5
+    - contents: "{{ pillar.host_id }}.open-contracting.org"
+
+# The Salt system.networking state does not fully support Ubuntu 20.04 yet so we are using cmd.run instead.
```

Comment:
Can you provide a link to an issue?

Reply:
Aha, https://github.com/open-contracting/deploy/issues/278#issuecomment-924485063.